src/dbnode/storage/index.go

```diff
@@ -1371,6 +1371,20 @@ func (i *nsIndex) queryWithSpan(
 		return false, err
 	}
 
+	if exhaustive {
+		i.metrics.queryExhaustive.Inc(1)
+	} else {
+		i.metrics.queryNonExhaustive.Inc(1)
+
+		if opts.RequireExhaustive {
+			return false, fmt.Errorf("query matched too many time series: require_exhaustive=%v, limit=%d, matched=%d",
+				opts.RequireExhaustive,
```

Comment:
Can we add a unit test here that tests when opts.RequireExhaustive is true it correctly returns an error on too many results?

Reply:
I think I added cases to cover this behavior in index_block_test.go - is there something missing from those tests changes that you're thinking of?